cmd/kueuectl/app/create/create_clusterqueue.go

```diff
@@ -0,0 +1,491 @@
+	case borrowingLimit:
+		rq.BorrowingLimit = ptr.To(quantity)
+	case lendingLimit:
+		rq.LendingLimit = ptr.To(quantity)
+	}
+
+	return rq, nil
+}
+
+func parseKeyValue(str, sep string) (string, string) {
+	pair := strings.SplitN(str, sep, 2)
+	if len(pair) == 1 {
+		return strings.TrimSpace(pair[0]), ""
+	}
+	return strings.TrimSpace(pair[0]), strings.TrimSpace(pair[1])
+}
+
+func mergeResourcesByFlavor(resourceGroups []v1beta1.ResourceGroup) ([]v1beta1.ResourceGroup, error) {
+	var mergedResources []v1beta1.ResourceGroup
+
+	indexByFlavor := make(map[string]int)
+	var index int
+	for _, rg := range resourceGroups {
+		flavorName := string(rg.Flavors[0].Name)
+		idx, found := indexByFlavor[flavorName]
+		if !found {
+			mergedResources = append(mergedResources, rg)
+			indexByFlavor[flavorName] = index
+			index++
+			continue
+		}
+
+		var err error
+		mergedResources[idx].Flavors[0].Resources, err = mergeResourceQuotas(mergedResources[idx].Flavors[0].Resources, rg.Flavors[0].Resources)
+		if err != nil {
+			// multiple FlavorQuotas with same name have been found but resources listed don't match
+			return mergedResources, errInvalidFlavor
+		}
+	}
+
+	return mergedResources, nil
+}
+
+func mergeResourceQuotas(rQuotas1, rQuotas2 []v1beta1.ResourceQuota) ([]v1beta1.ResourceQuota, error) {
+	var mergedResourceQuotas []v1beta1.ResourceQuota
+
+	for _, rq1 := range rQuotas1 {
+		idx := slices.IndexFunc(rQuotas2, func(rq v1beta1.ResourceQuota) bool { return rq.Name == rq1.Name })
+		if idx == -1 {
+			// both ResourceQuota lists should contain exactly the same resource names
+			return mergedResourceQuotas, errResourceQuotaNotFound
+		}
+
+		rq2 := rQuotas2[idx]
+		if rq1.NominalQuota.IsZero() {
+			rq1.NominalQuota = rq2.NominalQuota
+		}
+		if rq1.BorrowingLimit == nil {
+			rq1.BorrowingLimit = rq2.BorrowingLimit
+		}
+		if rq1.LendingLimit == nil {
+			rq1.LendingLimit = rq2.LendingLimit
+		}
+
+		mergedResourceQuotas = append(mergedResourceQuotas, rq1)
+	}
+
+	return mergedResourceQuotas, nil
+}
+
+func mergeFlavorsByCoveredResources(resourceGroups []v1beta1.ResourceGroup) ([]v1beta1.ResourceGroup, error) {
+	var mergedResources []v1beta1.ResourceGroup
+
+	indexByResourceGroupID := make(map[string]int)
+	var index int
+	for _, rg := range resourceGroups {
+		resourcesGroupID := getResourcesGroupID(rg.CoveredResources)
+		if idx, found := indexByResourceGroupID[resourcesGroupID]; found {
+			mergedResources[idx].Flavors = append(mergedResources[idx].Flavors, rg.Flavors...)
+			continue
+		}
+
+		if !isResourceGroupValid(indexByResourceGroupID, resourcesGroupID) {
+			return mergedResources, errInvalidResourceGroup
+		}
+		mergedResources = append(mergedResources, rg)
+		indexByResourceGroupID[resourcesGroupID] = index
+		index++
+	}
+
+	return mergedResources, nil
+}
+
+func getResourcesGroupID(coveredResources []corev1.ResourceName) string {
+	var s []string
+	for _, cr := range coveredResources {
+		s = append(s, string(cr))
+	}
```

Comment:
```suggestion
	 s := slices.Map(coveredResources , func(rn *corev1.ResourceName) string {return string(*rn)} )
```
https://github.com/kubernetes-sigs/kueue/blob/05a457eea1cef1066eafa3a5d6105c85c9773da2/pkg/util/slices/slices.go#L54

Reply:
Done